lib/fetch/common.c

```diff
@@ -639,8 +641,6 @@ happy_eyeballs_connect(struct addrinfo *res0, int verbose)
 			rv = -1;
 			break;
 		}
-		if (!waiting)
-			break;
 		else if (done)
```

Comment:
I think this can be removed, this is another bug.
If it tried all addresses done is 1. When there are >1 sockets it is waiting for, in the last long 300 second timeout and then one of the >1 sockets fails to connect i.e. with ECONNREFUSED, it will just stop waiting for the other sockets that may be established later.

Reply:
It makes a difference if all addresses fail to connect. In this case, without the check it would wait for the timeout while with the check, it fails immediately.
I think this fast failure is useful.

Reply:
But as is is, its dead code. It needs to be `if (done)`.

Not sure if fast failure is good. If there are two addresses, one ipv4 and one ipv6 and they are both too slow to respond within the connection delay, it will set done=1 and then set the timeout to 300 seconds.
If now one of them fails the other one would be closed.

Reply:
oops, I was referring to the `if(!waiting)` which I moved on top.